src/3d/Binding.fst

```diff
@@ -110,7 +103,19 @@ let nullary_macro t = { macro_arguments_t = []; macro_result_t = t }
 /// global_env maps top-level identifiers to their corresponding declaration
 ///  -- maps type identifiers to decl_attributes
 ///  -- maps macro names to their types
-let global_env = H.t ident' (decl & either decl_attributes macro_signature)
+noeq
+type global_env = {
+  ge_h: H.t ident' (decl & either decl_attributes macro_signature);
+  ge_fd: field_num_ops_t;

```

Comment:
Looks like the field num table is now carried as part of the environment. Sounds good, in general. Can you say a bit about what motivated this change? Would also be good to update the comment on the global_env type just above.

Reply:
This class of changes is to make sure that the field table and other global state is not shared among different .3d files provided on the command line.